### ISSUE TITLE
CORS issue upon clicking on One Signal push notification (web push)

### ISSUE THREAD

**Issue by varunpottigari** [OneSignal/OneSignal-Website-SDK#487]
I have integrated my angular 6 app with One Signal Push notifications. My index html file has the below code in the script tag.

var OneSignal = window.OneSignal || [];
    OneSignal.push(function () {
        OneSignal.init({
            appId: "xxxxxxxxxxxxxxxxx",
        });
    });

    OneSignal.push(['addListenerForNotificationOpened', function (data) {
        console.log(data);
    }]);
One Signal settings: The site is not fully HTTPS.

I hosted the web app on IIS. I was able to subscribe to notifications, send the notifications to the targetted user and was able to perform the appropriate action upon clicking the notification. The same action was tested multiple times on different browsers and devices.

After testing the app, the domain name of the website was changed and the appropriate changes in the settings have been made. (Only the domain name has been changed, the server where it is hosted in is the same)

After the change, the user can still subscribe to notifications and we can target the specific user to whom a notification has to be sent. But as the notification appears on the device, upon clicking it I am getting the below error logged onto the console and the listener is not being called.

Error: Access to fetch at 'https://onesignal.com/api/v1/notifications/{some one signal key}' from origin 'https: {one signal label for the website}' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Uncaught (in promise) TypeError: Failed to fetch OneSignalSDKWorker.js?appId={xxxxxxxxxxxxxx}

I am getting the same error across all the browsers.

The native versions of the mobile apps (Android (Kotlin) and iOS (Swift)) work perfectly fine before and after the change in URL.

I have tried changing the one signal settings: disables the site is not fully HTTPS toggle and then uploaded the OneSignalSDKWorker.js and OneSignalUpdaterWorker.js, manifest.json to the root directory and they are accessible without any redirects on the respective URLs where they have to be.

I have also tried by creating a new app in One Signal and changing the configurations on both the front end and one signal settings.

I have created a small test angular 6 application (to make sure that the app isn't creating the problem), integrated with one signal and deployed it in another server(to see if the server or the IIS is the cause) with a different URL (not the 2 that I used before).

Deleted the cache and cookies as well before trying any new change.

Nothing that I tried seems to resolve the issue.

Finally, I have checked another website that uses One Signal web push. I can see the same error logged into the console upon clicking on the notification.

I am not sure if the error getting logged is the expected behavior (didn't notice it before) and it is stopping from the listener getting triggered.

Also, I am thinking that changing the URL might have been a coincidence because I did not test for the click actions 1 week before and 2 days after changing the URL.

Need help in resolving this issue. Thanks in advance.

**Comment by rgomezp** [comment]
Hello,
You will need to enable CORS for what you are trying to do, you can read about this here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Comment by rgomezp** [comment]
Closing due to no response

**Comment by varunpottigari** [comment]
Hi team,

Here is the gif of the exact issue that I am facing. This is the code that I have inserted in the "script" tag of my index.html file. 

      ` var OneSignal = window.OneSignal || [];
        OneSignal.push(function () {
            OneSignal.init({
                appId: "xxxx-xxxx-xxx-xxxx-xxx-xxx",
            });
            console.log('One signal initiated.');
        });

         OneSignal.push(['addListenerForNotificationOpened', function (data) {
            console.log('Notification Clicked');
        }]);

        OneSignal.push(function () {
            OneSignal.on('notificationDisplay', function (event) {
                console.log('Notifiation Displayed');
            });
        });

        OneSignal.push(function () {
            OneSignal.on('notificationDismiss', function (event) {
                console.log('Notification Dismissed');
            });
        });`

Once the page is loaded, OneSignal.init() is called. 
When a notification is displayed, "notificationDisplay" event is triggered.
When a notification is dismissed, "notificationDismiss" event is triggered.
But when a notification is clicked, "addListenerForNotificationOpened" function is not called. But the documentation says that it should. (link [here](https://documentation.onesignal.com/docs/web-push-sdk#section--addlistenerfornotificationopened-))

It used to work previously about 2 months back, but stopped working after that. Can you please look into this and let me know if there is anything else that I have to do.

Thanks in advance.

**Comment by jkasten2** [comment]
@varunpottigari Thanks for reporting, however this doesn't look related to the CORS issue. Can you open an new issue?

**Comment by varunpottigari** [comment]
Hi @jkasten2 . I have created a new issue [here](https://github.com/OneSignal/OneSignal-Website-SDK/issues/507). Can you please look into it?

**Comment by rgomezp** [comment]
Hello @varunpottigari ,
I was able to reproduce the issue but are still investigating the cause of it. Great job catching this!

**Comment by rgomezp** [comment]
Hello everyone,
This issue should be fixed. Thanks for reporting it!

**Comment by erikramalho** [comment]
Guys for me it stills with problems. Do I have to update any file to get it working?

**Comment by erikramalho** [comment]
Hi @rgomezp I'm seeing this CORB Cross-Origin Read Blocking:
![Captura de Tela 2019-04-08 às 17 32 59](https://user-images.githubusercontent.com/5274068/55755516-931dbc00-5a25-11e9-8d76-c2882b18b62f.png)

This print was taken 2 minutes ago from a message sent from the dashboard.

**Comment by rgomezp** [comment]
see #507
